pkg/services/ngalert/notifier/external_alertmanager.go

```diff
@@ -0,0 +1,159 @@
+package notifier
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/url"
+
+	httptransport "github.com/go-openapi/runtime/client"
+	"github.com/grafana/grafana/pkg/infra/log"
+	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	amclient "github.com/prometheus/alertmanager/api/v2/client"
+)
+
+type externalAlertmanager struct {
+	log           log.Logger
+	url           string
+	tenantID      string
+	orgID         int64
+	amClient      *amclient.AlertmanagerAPI
+	httpClient    *http.Client
```

Comment:
I believe we already have a client requesting this API (I take it this is the config API) somewhere in Grafana (is what we use to proxy to external alertmanagers) - let's find it and abstract it. You don't have to do it as part of this PR but please file an issue for it.

With that I'd remove this client altogether.

Reply:
Done -> https://github.com/grafana/grafana/issues/74869